chemicalx/models/deepddi.py

```diff
@@ -1,14 +1,90 @@
 """An implementation of the DeepDDI model."""
 
-from .base import UnimplementedModel
+import torch
 
+from chemicalx.data import DrugPairBatch
+from chemicalx.models import Model
 __all__ = [
     "DeepDDI",
 ]
 
 
-class DeepDDI(UnimplementedModel):
+class DeepDDI(Model):
     """An implementation of the DeepDDI model.
 
     .. seealso:: https://github.com/AstraZeneca/chemicalx/issues/2
     """
+
+    def __init__(
+        self,
+        *,
+        drug_channels: int,
+        hidden_channels: int = 2048,
+        hidden_layers_num: int = 9,
+        out_channels: int = 1,
+    ):
+        """Instantiate the DeepDDI model.
+
+        :param drug_channels: The number of drug features.
+        :param hidden_channels: The number of hidden layer neurons.
+        :param hidden_layers_num: The number of hidden layers.
+        :param out_channels: The number of output channels.
+        """
+
+        super(DeepDDI, self).__init__()
+        assert hidden_layers_num > 1
+        dnn = []
+        dnn.extend([
+            torch.nn.Linear(drug_channels * 2, hidden_channels),
+            torch.nn.ReLU(),
+            torch.nn.BatchNorm1d(
+                num_features=hidden_channels,
+                affine=True,
+                momentum=None
+            ),
+            torch.nn.ReLU()
+        ])
+        for _ in range(hidden_layers_num - 1):
+            dnn.extend([
```

Comment:
just append the element instead of extending on a list

Reply:
Did you mean append each element separately? :-)

Reply:
Sorry I misread this part in the github viewer, on a second way the way you did it was very nice!

Reply:
That's OK! Thank you! :-)